Objects/Sensor.py

```diff
@@ -61,17 +88,45 @@ def __init__(self, monitor_num, box=None):
             
     def grab_data(self):
         img = self.sct.grab(self.box)
-        return np.array(img)
+        return np.array(img)[...,:3].astype(float)
+    
+    def trigger(self):
+        pw_abs = abs(self.grab_data() - self.mask).astype(np.uint8)
+        bool_mask =  (self.mask == self.background).all(axis=2,  keepdims=True).astype(int)
+        bool_mask = np.ones(bool_mask.shape) - bool_mask
+        mean_sum = np.mean(bool_mask * pw_abs)
+        print(mean_sum)
+        return mean_sum < self.treshold
+    
+    
     
 #Example
-s = ImageSensor(1, (720,100,1000,500))
+#(left, top, widht, height)
+gta_mask = cv2.imread("C:/Users/Santiago/Pictures/gta_stars_mask.png")
+s = MaskSensor(1, mask = gta_mask, treshold = 1, box = (1861,9,34,47))
+
+time_check = 3
 
 while True:
-    img = s.grab_img()
-    cv2.imshow('screen', img)
- 
+    #img = s.grab_data().astype(np.uint8)
```

Comment:
Commented code should not exist on a repository. If we need the code, we can look back at the repository history